src/actions/index.js

```diff
@@ -0,0 +1,81 @@
+import axios from 'axios'
+import axiosWithAuth from '../axiosAuth'
+
+export const ALREADY_LOGGED_IN = 'ALREADY_LOGGED_IN'
+export const SEARCH_SUCCESS = 'LOGGING_IN'
+export const CLEAR_ERRORS = 'CLEAR_ERRORS'
+export const LOGGING_IN = 'LOGGING_IN'
+export const LOGIN_SUCCESS = 'LOGIN_SUCCESS'
+export const LOGIN_FAILURE = 'LOGIN_FAILURE'
+export const REGISTERING = 'REGISTERING'
+export const REGISTER_SUCCESS = 'REGISTER_SUCCESS'
+export const REGISTER_FAILURE = 'REGISTER_FAILURE'
+export const PASSWORD_MISMATCH = 'PASSWORD_MISMATCH'
+export const LOGOUT = 'LOGOUT'
+
+  export const passwordMismatch = () => dispatch => {
+    dispatch({
+      type: PASSWORD_MISMATCH
+    })
+  }
+
+export const ensureLoggedIn = () => dispatch => {
+  dispatch({
+    type: ALREADY_LOGGED_IN
+  })
+}
+
+export const clearErrors = () => dispatch => {
+  dispatch({
+    type: CLEAR_ERRORS
+  })
+}
+
+export const logout = () => dispatch => {
+  dispatch({
+    type: LOGOUT
+  })
+}
+
+export const login = (user, pass) => dispatch => {
+  dispatch({
+    type: LOGGING_IN
+  })
+  axios.post('https://bw-spotify-backend.herokuapp.com/api/login', {username: user, password: pass})
```

Comment:
Im assuming this is copy and pasted and you're aware it needs to be changed to our links.